src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -260,112 +278,112 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
-
-                                // Check if a change has been detected in the key-value registered for refresh
-                                if (change.ChangeType != KeyValueChangeType.None)
+                                refreshAll = await RefreshIndividualKvWatchers(
+                                    client,
+                                    keyValueChanges,
+                                    refreshableIndividualKvWatchers,
+                                    endpoint,
+                                    logDebugBuilder,
+                                    logInfoBuilder,
+                                    cancellationToken).ConfigureAwait(false);
+
+                                if (!refreshAll)
                                 {
-                                    logDebugBuilder.AppendLine(LogHelper.BuildKeyValueReadMessage(change.ChangeType, change.Key, change.Label, endpoint.ToString()));
-                                    logInfoBuilder.AppendLine(LogHelper.BuildKeyValueSettingUpdatedMessage(change.Key));
-                                    keyValueChanges.Add(change);
-
-                                    if (changeWatcher.RefreshAll)
-                                    {
-                                        refreshAll = true;
-                                        break;
-                                    }
+                                    // Get feature flag changes
+                                    watchedFfCollections = new Dictionary<KeyValueSelector, IEnumerable<MatchConditions>>(_watchedFfCollections);
+
+                                    await RefreshFfCollections(client,
+                                        keyValueChanges,
+                                        refreshableFfWatchers,
+                                        watchedFfCollections,
+                                        endpoint,
+                                        logDebugBuilder,
+                                        logInfoBuilder,
+                                        cancellationToken).ConfigureAwait(false);
                                 }
                                 else
                                 {
-                                    logDebugBuilder.AppendLine(LogHelper.BuildKeyValueReadMessage(change.ChangeType, change.Key, change.Label, endpoint.ToString()));
+                                    // Trigger a single load-all operation if a change was detected in one or more key-values with refreshAll: true
+                                    data = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
+                                    watchedKvCollections = await LoadSelected(client, data, _options.KeyValueSelectors, cancellationToken).ConfigureAwait(false);
+                                    watchedFfCollections = await LoadSelected(client, data, _options.FeatureFlagSelectors, cancellationToken).ConfigureAwait(false);
```

Comment:
Is this by design that a watched setting can trigger feature flag refresh? I suppose watched setting will only affect key values. Feature flag refresh has no business with key value refresh.

Reply:
Is this because snapshot may contain feature flags?